lib/launchers/amz.rb

```diff
@@ -471,6 +480,89 @@ def get_regions
       client_ec2.describe_regions.to_a[0][0]
     end
 
+    def default_zone
+      config[:install_base_domain].sub(/\.$/,"") + "."
+    end
+
+    def default_zone_id
+      @default_r53_zone_id ||= r53_zone_id_by_domain(default_zone)
+    end
+
+    def r53_zone_id_by_domain(domain)
+      zone = client_r53.list_hosted_zones.hosted_zones.find { |z|
+        z.name == domain
+      }
+      unless zone
+        raise "can't find zone '#{domain}' in AWS"
+      end
+      return zone.id.sub(%r{.*/(.*)$}, "\\1")
+    end
+
+    def r53_zone_by_id(id)
+      zone = client_r53.get_hosted_zone({id: id})
+      return zone.hosted_zone.id.sub(%r{.*/(.*)$}, "\\1")
+    end
+
+    # @param [Hash] changes might be something like
+    #   [{
+    #     :action=>"UPSERT",
+    #     :resource_record_set=> {
+    #       :name=>"myhost.qe.devcluster.openshift.com",
+    #       :resource_records=> [{:value=>"192.0.2.44"}],
+    #       :ttl=>60,
+    #       :type=>"A"
+    #     }
+    #   }]
+    # @see https://docs.aws.amazon.com/sdk-for-ruby/v2/api/Aws/Route53/Client.html#change_resource_record_sets-instance_method
+    def change_resource_record_sets(zone_id: nil, changes:)
+      zone_id ||= default_zone_id
+      client_r53.change_resource_record_sets(
+        hosted_zone_id: zone_id,
+        change_batch: { changes: changes }
+      )
+    end
+
+    def list_resource_record_sets(zone_id: nil)
+      zone_id ||= default_zone_id
+      res = []
+      client_r53.list_resource_record_sets(hosted_zone_id: zone_id).each_page { |p|
+        res.concat p.resource_record_sets.map(&:to_h)
+      }
+      return res
+    end
+
+    # @param [Regexp] re
+    # @note be very careful to avoid interfering regular expressions
+    # example: delete_resource_records_re(/my-hostname/)
+    def delete_resource_records_re(re, zone_id: nil)
+      logger.warn("Removing Route53 records matching #{re.inspect}")
+      records = list_resource_record_sets(zone_id: zone_id)
+      to_delete = records.select { |r| r[:name] =~ re }
+      logger.debug("Removing records: #{to_delete.map {|r| r[:name]}}")
+      list = to_delete.map { |r| { action: "DELETE", resource_record_set: r } }
+      change_resource_record_sets(zone_id: zone_id, changes: list)
+    end
+
+    def create_a_records(name, ips, zone_id: nil, ttl: 180)
```

Comment:
Can we change the default TTL to 60(s)? This value is used by other team already and it is more legible for users I guess.

Reply:
TTL is time that caching DNS servers may keep record it received. It's not something user visible. I didn't look at OCP installation details, do you think that after creating the record, installer will need it immediately resolvable?

My assumption was that there will be several minutes before these DNS records matter at all. But if I'm mistaken, I'll be happy to change.

Reply:
Just to clarify, TTL only matters when removing or changing an existing record. Then if that record has been cached on some DNS server, the server will try to resolve it again (and see the new value) at most after TTL finished since name was resolved.

Reply:
Yes, it has to be present because installer depends on DNS resolving too - based on my measuring, it calls DNS for that ~5minutes after installer called but it depends on the power so it might be a bit sooner or later.

5 mins isn't a large timeframe, that's the reason I'd prefer to by in SYNC mode at least during DNS records creation.

Reply:
Ok, lets add sync mode in another PR. My understanding was that installer would retry for some time until things start working. But I don't mind having SYNC mode as we discussed above.

Reply:
Yes installer tries it until timeout (hard-coded 40minutes now if I recall correctly).